### DISCUSSION TITLE
Credential issuance portal demo

### DISCUSSION THREAD

**Discussion by pramodkvarma** [Sunbird-RC/community#196] [Ideas & Enhancements]
I feel we should have a credential issuer portal demo hosted by one of the community members. 

Basic capabilities I am imagining:
1. Issuers sign up
2. Choose a template (like in Google docs) from existing templates or upload an SVG template
3. Configure the template with logo/icons/colors
4. Save template
5. Generate 1 or many (via CSV upload) credentials against a saved template.
6. Issuers should be able to download the cred files.
7. Simulate "send via email/SMS" or "send via Digilocker" etc (just UI part for demo, need not actually send)
8. Use standard reference app to scan the QR code during demo

First of all, @parthlawate do we already have a reference portal like this that was built for eLocker use case? If so, leverage that to create this demo instance.

@surendrasinghs will send some more details about the demo use case. Please post doc link when you have it.

Anyone up for taking this up? I will be posting this URL on Discord as well. We will use this thread to discuss and work on this item.

**Comment by parthlawate** [comment]
This has been on the todo for the elocker case and the detailed stories for
these are in place and the development on this piece has started .

Will plan to spin up a reference demo once ready.

**Reply by pramodkvarma** [reply]
Perfect, thanks! We will need this on some urgent basis. Any possibility of fast tracking?

@surendrasinghs can you study what's already there and align with it to avoid additional software effort? Thanks.

**Reply by surendrasinghs** [reply]
@parthlawate Studied https://github.com/orgs/Sunbird-RC/projects/2. Couldn't find the "issue VC workflow" there.

I propose/imagine adding a step in the basic capabilities @pramodkvarma mentioned above - i.e. to select 'What do you want to issue? Proof of work/training/etc'. This should come after (step 1) sign-up/log-in and before (step 3) selecting or uploading a template.

Here's the basic set of capabilities I am imagining

1. Sign-up or Log in as Issuer
2. **Select what do you want to issue - Proof of Work/Reputation/Training/Skill**
3. Choose a template (like in Google docs) from existing templates or upload an SVG template with placeholder variables
4. Configure template (change logos, colours, upload background image, signatures, etc)
5. Save template
6. Start issuing VC - enter details or bulk upload (in future) or bulk issue UI (in future)
7. Issuers should be able to download the cred files.
8. Simulate "send via email/SMS" or "send via Digilocker" etc (just UI part for a demo, need not actually send)
9. Use a standard reference app to scan the QR code during the demo

**Reply by pramodkvarma** [reply]
Response from @tejash-jl on Discord Sunbird-RC channel (pasting below):

we have a similar reference implementation at https://github.com/dileepbapat/ref-sunbirdrc-certificate
we have created a jupyter notebook which can be used to trigger apis from signup till generating a certificate(pdf/html/svg) and also contains a sample UI to verify the certificate

@parthlawate @surendrasinghs FYI

Btw can you both join Discord channel as well?

**Reply by parthlawate** [reply]
Yes I'm on Discord already. Discussed with @tejash-jl . We will be building the UI on top pf this. We had done the UI a flow for the elocker, we are modifying the mocks for this flow. Will post here once ready.

**Reply by pramodkvarma** [reply]
Super! 👍

**Reply by parthlawate** [reply]
@surendrasinghs The detailed stories are not added on there yet. Will add those in the next couple of days.

**Comment by parthlawate** [comment]
Sure will plan to speed it up .

**Comment by shankarmaruwada** [comment]
That's great to hear, Parth! 👍

**Comment by jayadinni** [comment]
Mock for Issuer Portal:
1. It has mock html and schema2. 
2. Assumptions: User (issuer) can choose available template and edit if required.
3. Next Sprint/Releases: SVG upload, Download saved Credentials

Onboarding and Setting up VC:


https://user-images.githubusercontent.com/580711/156002365-4f11e210-9241-4784-9bef-b7ac318bf512.mp4

**Reply by pramodkvarma** [reply]
Excellent start @jayadinni . Can you look at the earlier comments on this discussion and see how "templates" can be configured, and design the issuer "home page" where they can see templates, issued certs, etc. 

@surendrasinghs please get on a call with @parthlawate and @jayadinni to create next version of this please.

**Reply by ChakshuGautam** [reply]
@jayadinni can you share the Github repo for the demo portal?

**Reply by parthlawate** [reply]
@Renuka-S @Pratikshakhandagale @code-blooded-guy can you help with this ?

**Reply by Pratikshakhandagale** [reply]
@ChakshuGautam Here is the Github repo link for issuance demo portal 
- UI code : [here](https://github.com/code-blooded-guy/sunbird-rc-ui/tree/issuance)
- UI config files : [https://github.com/Sunbird-RC/demo-certificate-issuance](url)

**Comment by jayadinni** [comment]
Should we allow issuer to configure two templates for a VC .. one is native and other print version?

**Reply by pramodkvarma** [reply]
To start with use SVG template that will be both digital and print capable. Will enhance later to allow separate digital and print template configuration per VC type. Issuer can setup multiple named VC types.

**Comment by amarts** [comment]
Few questions on this thread.

What are the criteria for listing a tool  as a 'Demo' Issuer flow product? Asking because we have a SaaS product which issues VCs hosted at https://studio.dhiway.com

We do have a 30 days free trial for anyone who wants to use it.

Happy to work with the community to build a flow to meet complete flow with a possible guest mode.

**Reply by pramodkvarma** [reply]
That's great @amarts. We can definitely list that as part of the "VC spec" community project. 

This thread was specifically about a VC issuance demo using Sunbird-RC setup by the community for experiencing Sunbird-RC. Hope that answers.

**Comment by pramodkvarma** [comment]
@parthlawate @surendrasinghs any update on this? Can we bring up a demo instance?

**Reply by parthlawate** [reply]
Work is in progress. Will try to commit some dates in the next couple of days for a first cut demo.

**Reply by pramodkvarma** [reply]
Early preview possible? Needed rapid iterations and hence requesting. Thanks a lot.

**Reply by parthlawate** [reply]
Planning for a first look this Friday..
